spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -769,5 +769,50 @@
         end
       end
     end
+
+    describe "gubernatorial elections fund" do
+      context "not mfj" do 
+        let(:intake) { 
+          create(:state_file_nj_intake)
+        }
+        it "does not fill the field when the answer is no" do 
+          expect(xml.at("Body PrimGubernElectFund")).to eq(nil)
+          expect(xml.at("Body SpouCuPartPrimGubernElectFund")).to eq(nil)
+        end
+
+        it "checks yes when the answer is yes" do
+          intake.primary_contribution_gubernatorial_elections = :yes
+          expect(xml.at("Body PrimGubernElectFund").text).to eq('X')
+        end
+      end
+
+      context "mfj" do 
+        context "when primary does not contribtue and spouse does" do 
+          let(:intake) { 
+            create(:state_file_nj_intake, 
+            :married_filing_jointly,
+            primary_contribution_gubernatorial_elections: :no, spouse_contribution_gubernatorial_elections: :yes,
+            )
+          }
+          it "leaves primary blank and marks an X for spouse" do 
+            expect(xml.at("Body PrimGubernElectFund")).to eq(nil)
+            expect(xml.at("Body SpouCuPartPrimGubernElectFund").text).to eq('X')
+          end
+        end
+
+        context "when primary wants to contribute and spouse does not" do 
+          let(:intake) { 
+            create(:state_file_nj_intake, 
+            :married_filing_jointly,
+            primary_contribution_gubernatorial_elections: :yes, spouse_contribution_gubernatorial_elections: :no,
+            )
+          }
+          it "checks yes for primary and no for spouse" do 
+            expect(xml.at("Body PrimGubernElectFund").text).to eq('X')
+            expect(xml.at("Body SpouCuPartPrimGubernElectFund")).to eq(nil)
+          end
```

Comment:
[dust] it could be nice to use the `each` pattern here that Matt demo'd on his recent PR to check all 4 cases